src/raw_rwlock.rs

```diff
@@ -898,41 +911,49 @@ impl RawRwLock {
         self.lock_upgradable();
     }
 
-    // Common code for waking up parked threads after releasing WRITER_BIT or
-    // UPGRADABLE_BIT.
+    /// Common code for waking up parked threads after releasing WRITER_BIT or
+    /// UPGRADABLE_BIT.
+    ///
+    /// # Safety
+    ///
+    /// `callback` must uphold the requirements of the `callback` parameter to
+    /// `parking_lot_core::unpark_filter`. Meaning no panics or calls into any function in
+    /// `parking_lot`.
     #[inline]
-    fn wake_parked_threads<C>(&self, new_state: usize, callback: C)
-    where
-        C: FnOnce(usize, UnparkResult) -> UnparkToken,
-    {
+    unsafe fn wake_parked_threads(
+        &self,
+        new_state: usize,
+        callback: impl FnOnce(usize, UnparkResult) -> UnparkToken,
```

Comment:
Not related to the safety. Went ahead and made this, and two more places, `impl Trait`s instead. They are just internal functions so should not affect the API.